File: Loenn/entities/sans_boss.lua
local drawableSprite = require("structs.drawable_sprite")
local utils = require("utils")

local sansBoss = {}

sansBoss.name = "MaggyHelper/SansBoss"
sansBoss.depth = -8500
sansBoss.justification = {0.5, 1.0}

sansBoss.placements = {
    {
        name = "sans_boss",
        data = {
            health = 1100,
            maxHealth = 1100,
        }
    }
}

sansBoss.fieldInformation = {
    health = {
        fieldType = "integer",
        minimumValue = 1
    },
    maxHealth = {
        fieldType = "integer",
        minimumValue = 1
    }
}

sansBoss.fieldOrder = {
    "x", "y",
    "health",
    "maxHealth"
}

function sansBoss.sprite(room, entity)
    local textures = {
        "characters/sans/idle00",
        "characters/player/sitDown00"
    }

    for _, texture in ipairs(textures) do
        local ok, sprite = pcall(drawableSprite.fromTexture, texture, entity)
        if ok and sprite then
            sprite:setJustification(0.5, 1.0)
            return {sprite}
        end
    end

    return {}
end

function sansBoss.selection(room, entity)
    return utils.rectangle(entity.x - 13, entity.y - 36, 26, 36)
end

return sansBoss

File: Ahorn/entities/SansBoss.jl
module SansBoss

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/SansBoss" SansBoss(x::Integer, y::Integer, health::Integer=1100, maxHealth::Integer=1100)

const placements = Ahorn.PlacementDict(
    "sans_boss" => Ahorn.EntityPlacement(SansBoss)
)

function Ahorn.selection(entity::SansBoss)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::SansBoss, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
